openghg/store/base/_base.py

```diff
@@ -166,30 +166,42 @@ def remove_datasource(self: T, uuid: str) -> None:
         """
         del self._datasource_uuids[uuid]
 
-    def add_datasources(self, uuids: Dict, data: Dict, metastore: TinyDB) -> None:
+    def add_datasources(self,
+                        uuids: Dict,
+                        data: Dict,
+                        metastore: TinyDB,
+                        update_keys: Optional[list] = None) -> None:
         """Add the passed list of Datasources to the current list
 
         Args:
             datasource_uuids: Datasource UUIDs
             metadata: Metadata for each species
             metastore: TinyDB metadata store
+            update_keys: List of keys to update within metastore for existing data.
         Returns:
             None
         """
         from openghg.util import to_lowercase
 
         for key, uuid_data in uuids.items():
+            meta_copy = data[key]["metadata"].copy()
+
+            uid = uuid_data["uuid"]
             new = uuid_data["new"]
-            # Only add if this is a new Datasource
+
+            # Add new entry if this is a new Datasource
             if new:
-                meta_copy = data[key]["metadata"].copy()
-                uid = uuid_data["uuid"]
                 meta_copy["uuid"] = uuid_data["uuid"]
 
                 # Make sure all the metadata is lowercase for easier searching later
                 meta_copy = to_lowercase(d=meta_copy)
                 metastore.insert(meta_copy)
                 self._datasource_uuids[uid] = key
+            # Update keys in metastore as appropriate
+            elif update_keys is not None:
+                update = {key: value for key, value in meta_copy.items() if key in update_keys}
+                q = Query()
+                metastore.upsert(update, q.uuid == uid)
```

Comment:
Ah yeah I recently looked into this again and I actually think this may be better as `update()`... I think `upsert()` would be good if we were adding a complete set of metadata at this point but as it's limited this may sometimes do slightly the wrong thing - see what you think https://tinydb.readthedocs.io/en/latest/usage.html#upserting-data

Reply:
Actually yeah, update might be what we want here.

Reply:
Ok I changed this from upsert to update

Reply:
OK great.